contracts/deploy/mainnet/117_oeth_fixed_rate_dripper.js

```diff
@@ -0,0 +1,101 @@
+const addresses = require("../../utils/addresses");
+const { deploymentWithGovernanceProposal } = require("../../utils/deploy");
+
+module.exports = deploymentWithGovernanceProposal(
+  {
+    deployName: "117_oeth_fixed_rate_dripper",
+    forceDeploy: false,
+    //forceSkip: true,
+    reduceQueueTime: true,
+    deployerIsProposer: false,
+    proposalId: "",
+  },
+  async ({ deployWithConfirmation, withConfirmation }) => {
+    const cOETHVaultProxy = await ethers.getContractAt(
+      "VaultAdmin",
+      addresses.mainnet.OETHVaultProxy
+    );
+
+    // Deployer Actions
+    // ----------------
+    const { deployerAddr } = await getNamedAccounts();
+    const sDeployer = await ethers.provider.getSigner(deployerAddr);
+
+    // 1. Upgrade Dripper to the new version (with transferAll token function)
+    // 2. Deploy OETH fixed rate dripper
+    // 3. Transfer all funds from old dripper to new dripper
+    // 4. Set new dripper on the vault
+
+    // --- 1 ---
+    // 1.a. Get the current OETH Dripper Proxy
+    const cOETHDripperProxy = await ethers.getContractAt(
+      "OETHDripperProxy",
+      "0xc0F42F73b8f01849a2DD99753524d4ba14317EB3"
+    );
```

Comment:
Any reason for not using `getContract` here? For proxies, we can just let hardhat get the last deployed address, it shouldn't be a problem